### ISSUE TITLE
The result is different from the huggingface?

### ISSUE THREAD

**Issue by yuhuangyue** [VAST-AI-Research/TripoSR#16]
I found that the result of run.py is different from the result displayed in huggingface. 
Are there any parameters that are different?

Following are these 2 different results.
![image](https://github.com/VAST-AI-Research/TripoSR/assets/27202239/6d6af9c0-7f96-4446-82f7-f53a843cd28f)

![image](https://github.com/VAST-AI-Research/TripoSR/assets/27202239/a51ef471-f9ae-4644-bc04-8c75cdf1c0ca)

**Comment by yuhuangyue** [comment]
![image](https://github.com/VAST-AI-Research/TripoSR/assets/27202239/df6f28f9-600c-4358-8a7d-a4f008f9cc5f)
![image](https://github.com/VAST-AI-Research/TripoSR/assets/27202239/b04caad3-e98e-456c-a7e4-51b70f7fefbf)

**Comment by pookiefoof** [comment]
It could be a pre-processing issue. The images under `examples` have been pre-processed (cut out the foreground and fill with gray background), so please just run with the `--no-remove-bg` option to get the same results as the HF demo.
We use [rembg](https://github.com/danielgatis/rembg) as a lightweight foreground extraction tool. Other tools such as the [Clipdrop](https://clipdrop.co/remove-background) may achieve better results.

**Comment by yuhuangyue** [comment]
It seems not the pre-processing issue, I run 

`python run.py examples/horse.png --output-dir output/ --no-remove-bg`

But the results still different from HF demo

**Comment by pookiefoof** [comment]
Could you provide the outputs so that we can look into them?

**Comment by yuhuangyue** [comment]
The output is the same as the images I post before.
There is no different output when I add `--no-remove-bg` option

**Comment by pookiefoof** [comment]
[mesh.obj.zip](https://github.com/VAST-AI-Research/TripoSR/files/14503838/mesh.obj.zip)
Just tested with `python run.py examples/horse.png --output-dir output/ --no-remove-bg` and I uploaded the output mesh, it seems to be consistent with the HF demo.

Can you provide more info about your machine environment?

**Comment by yuhuangyue** [comment]
I remembered that when I fix this bug, I changed the code here, and using cpu to extract mesh, Could it be because of this??

https://github.com/VAST-AI-Research/TripoSR/issues/3#issuecomment-1978104374

**Comment by pookiefoof** [comment]
That's a bit weird because using CPU marching cubes should not affect the results...
Have you tested on more images?

**Comment by yuhuangyue** [comment]
Yes, i also think it's weird, but using CPU marching cubes indeed affect the results.
I notice that you refined your code in isosurface.py to avoid torchmcubes was not compiled with CUDA support
```
    def forward(
        self,
        level: torch.FloatTensor,
    ) -> Tuple[torch.FloatTensor, torch.LongTensor]:
        level = -level.view(self.resolution, self.resolution, self.resolution)
        try:
            v_pos, t_pos_idx = self.mc_func(level.detach(), 0.0)
        except AttributeError:
            print("torchmcubes was not compiled with CUDA support, use CPU version instead.")
            v_pos, t_pos_idx = self.mc_func(level.detach().cpu(), 0.0)
        v_pos = v_pos[..., [2, 1, 0]]
        v_pos = v_pos / (self.resolution - 1.0)
        return v_pos.to(level.device), t_pos_idx.to(level.device)
```
But this will results different output as I post before. I have test.

The input image is 
![image](https://github.com/VAST-AI-Research/TripoSR/assets/27202239/6233a11d-953c-41a8-9d32-cb1c2924ae76)

when I run `python run.py output/13/13.png --output-dir output/`, it turn out
![image](https://github.com/VAST-AI-Research/TripoSR/assets/27202239/ff31e725-6b58-4978-bb70-3ecf73ddafb2)

when I run `python run.py output/13/13.png --output-dir output/ --device cpu`, it turn out
![image](https://github.com/VAST-AI-Research/TripoSR/assets/27202239/5d3849b0-17a9-4a9d-a25d-1ff87a7718d0)

It's weird.
